### PR TITLE
Fix update command

### PR DESCRIPTION
**Pull request by Arminek**

### CODE REVIEW

File: src/Command/UpdateSimpleProducts.php [hunk 1]
```diff
@@ -31,18 +42,24 @@
...
      * @param Manager $manager
      * @param ProductDocumentFactoryInterface $productDocumentFactory
+     * @param LoggerInterface $logger
      */
-    public function __construct(
-        ProductRepositoryInterface $syliusProductRepository,
-        Manager $manager,
-        ProductDocumentFactoryInterface $productDocumentFactory
-    ) {
+    public function __construct(ProductRepositoryInterface $syliusProductRepository, ChannelRepositoryInterface $syliusChannelRepository, Manager $manager, ProductDocumentFactoryInterface $productDocumentFactory, LoggerInterface $logger)
```
Comment (lchrusciel) [inline src/Command/UpdateSimpleProducts.php:56]: Could be split into multiple lines

File: src/Command/UpdateSimpleProducts.php [hunk 2]
```diff
@@ -63,42 +81,101 @@ protected function configure()
...
-        });
+        /** @var Channelinterface[] $channels */
+        $channels = $this->syliusChannelRepository->findAll();
 
-        $productDocumentCreatedCount = 0;
-        /** @var ProductInterface[] $syliusSimpleProducts */
-        foreach ($syliusSimpleProducts as $syliusProduct) {
-            $channels = $syliusProduct->getChannels();
+        foreach ($channels as $channel) {
+            $locales = $channel->getLocales();
```
Comment (lchrusciel) [inline src/Command/UpdateSimpleProducts.php:92]: Body of this for each could go to separate private class, just to follow object calisthenics

File: src/Command/UpdateSimpleProducts.php [hunk 3]
```diff
@@ -63,42 +81,101 @@ protected function configure()
...
+        OutputInterface $output
+    ) {
+        $progress = new ProgressBar($output, $howManyProductsToCreate);
+        $progress->start();
+        $iterations = ceil($howManyProductsToCreate / $fetchLimit);
+        $lastPersistedProductId = 0;
+
+        for ($i = 0; $i < $iterations; $i++) {
+            /** @var ProductInterface[] $syliusProducts */
+            $syliusProducts = $this->syliusProductRepository
```
Comment (lchrusciel) [inline src/Command/UpdateSimpleProducts.php:127]: Can we build most of the query outside this loop and only call 
```php
$this->fetchProductsQuery()
    ->setParameter('lastPersistedProductId', $lastPersistedProductId)
    ->getQuery()
    ->getResult()
;
```

File: src/Command/UpdateSimpleProducts.php [hunk 4]
```diff
@@ -63,42 +81,101 @@ protected function configure()
...
+     * @param ChannelInterface $channel
+     * @param LocaleInterface $locale
+     * @param int $fetchLimit
+     * @param int $howManyProductsToCreate
+     * @param OutputInterface $output
+     */
+    private function createProducts(
+        ChannelInterface $channel,
+        LocaleInterface $locale,
+        $fetchLimit,
```
Comment (lchrusciel) [inline src/Command/UpdateSimpleProducts.php:116]: Missing typehints and return types

Comment (lchrusciel) [reply src/Command/UpdateSimpleProducts.php:116]: In the whole class

File: tests/Command/UpdateSimpleProductsTest.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\ElasticSearchPlugin\Command;
+
+use Lakion\ApiTestCase\JsonApiTestCase;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+
```
Comment (lchrusciel) [inline tests/Command/UpdateSimpleProductsTest.php:11]: Redundant blank line

File: tests/Command/UpdateSimpleProductsTest.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        $command = $application->find('sylius:elastic-search:update-simple-product');
+
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(['command' => $command->getName()]);
+    }
+
+    /**
+     * @param int $expected
+     */
+    private function assertElasticSearchDocumentCount($expected)
```
Comment (lchrusciel) [inline tests/Command/UpdateSimpleProductsTest.php:47]: missing type hints

File: tests/Command/UpdateSimpleProductsTest.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        $application = new Application(static::$sharedKernel);
+        $application->add(static::$sharedKernel->getContainer()->get('sylius_elastic_search.command.update_product'));
+        $command = $application->find('sylius:elastic-search:update-simple-product');
+
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(['command' => $command->getName()]);
+    }
+
+    /**
+     * @param int $expected
```
Comment (lchrusciel) [inline tests/Command/UpdateSimpleProductsTest.php:45]: Also, we can remove this docblock if we will use typehints

**Review by lchrusciel — APPROVED** [review]

**Comment by pamil** [comment]
We need a command that does not remove existing index :D

**Pull request closed without merging**